packages/document-model-libs/editors/document-model-2/editor.tsx

```diff
@@ -59,18 +52,30 @@ export default function Editor(
       document.state.global.specifications[0].modules,
     ],
   );
-
-  const schema = buildSchema(`
-    ${hiddenQueryTypeDefDoc}
-    ${globalStateSchema}
-    ${localStateSchema}
-    ${modules
-      .flatMap((module) =>
-        module.operations.map((operation) => operation.schema),
-      )
-      .filter(Boolean)
-      .join("\n")}
-  `);
+  const [schema, setSchema] = useState(initialSchema);
+  const [errors, setErrors] = useState("");
+
+  useEffect(() => {
+    try {
+      const newSchema = buildSchema(`
```

Comment:
`buildSchema` throws on invalid input. use try catch and show errors in component